src/snovault/vis_defines.py

```diff
@@ -0,0 +1,1614 @@
+        #                                         "nuclear RNA", "genomic DNA", "protein", "other"]
+        #         "disease": "...",              optional?  # TODO
+        #         "disease_ontology_uri": "...", optional?  # TODO
+        #         "biomaterial_type": "...",     ["Cell Line", "Primary Cell", "Primary Cell Culture",
+        #                                         "Primary Tissue"]
+        #         "line":                        biosample_term_name
+        #         "lineage":                     ??? HELPME: Where do I get this?
+        #         "differentiation_stage":       ??? HELPME: Where do I get this?
+        #         "medium":                      ??? HELPME: Where do I get this?
+        #         "sex":                         experiment.replicate.library.biosample.sex
+        #     },
+        #     "sample_id_2": { ... }
+        # }
+        ihec_json["samples"] = self.samples
+
+        datasets = {}
+        # "datasets": {
+        #    "experiment_1": {    one per experiment    accession
+        #        "sample_id": "...",                    biosample_term
+        #        "experiment_attributes": {
+        #            "experiment_type": "...",
+        #            "assay_type": "...",               assay_term_name  Match ontology URI
+        #                                                           (e.g. 'DNA Methylation')
+        #            "experiment_ontology_uri": "...",  assay_term_id (e.g. OBI:0000716)
+        #            "reference_registry_id": "..."     IHEC Reference Epigenome registry ID,
+        #                                                 assigned after submitting to EpiRR
+        #        },
+        #        "analysis_attributes": {
+        #            "analysis_software": "...",           pipeline
+        #            "analysis_software_version": "..."    pipeline version
+        #            "analysis_group": "...",              pipeline laboratory
+        #            "alignment_software": "...",          software ugly lookup. Single for whole experiment!
+        #            "alignment_software_version": "...",  software_version
+        #        },
+        #        "browser": {
+        #            "signal_forward": [               view
+        #                {
+        #                    "big_data_url": "...",    obvious
+        #                    "description_url": "...", Perhaps not
+        #                    "md5sum": "...",          Add this to metadata pairs?
+        #                    "subtype": "...",         More details?
+        #                    "sample_source": "...",   pooled,rep1,rep2
+        #                    "primary":                pooled or rep1 ?
+        #                },
+        #                { ... }
+        #            ],
+        #            "signal_reverse": [ { ... } ]
+        #        }
+        #    },
+        #    "experiment_2": {
+        #        ...
+        #    },
+
```

Comment:
Spoke to Karthik about adding the vis_defines.py file.  This is an unfortunate dependency in helpers.py on vis_format_url without a clear solution.  We'll want to examine this in more detail.  Probably resolve/merge that ticket, ENCD-4116, first and then integrate it with search.  It would be a bad idea to hide that change in the search update ticket. 

notes:
-Copying the whole file into snovault is redundant and confusing, but the easiest.  
-Moving the whole file to snovault would split the visualization code.    
-We definitely don't want snovault dependent on encoded either.

Reply:
@caseylitton I addressed ENCD-4116, a while ago. I recommend thinking about deploying that after this current PR is settled.